src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -378,9 +394,28 @@ private void ChangeThreadCount(int newThreadCount, StateOrTransition state)
                 _completionsSinceLastChange = 0;
             }
 
-            private void LogTransition(int newThreadCount, double throughput, StateOrTransition state)
+            private void LogTransition(int newThreadCount, double throughput, StateOrTransition stateOrTransition)
             {
-                // TODO: Log transitions
+                // Use the _log array as a circular array for log entries
+                int index = (_logStart + _logSize) % LogCapacity;
+
+                if(_logSize == LogCapacity)
+                {
+                    _logStart = (_logStart + 1) % LogCapacity;
+                    _logSize--; // hide this slot while we update it
```

Comment:
This isn't for thread safety. It's for always having `_logSize` correctly be the number of elements in the log that are currently accurate. We decrement `_logSize` since we are editing the formerly first and now last element in our `_log` array since our log is already at capacity. This also simplifies the code at the end of the method where we increment `_logSize` unconditionally because in all cases we are adding a new element to the log.